### DISCUSSION TITLE
Application Persistence

### DISCUSSION THREAD

**Discussion by AdmnJ** [ScoopInstaller/Extras#8534] [Q&A] [answered]
So a number of applications like web browsers for instance save bookmarks, extensions, ect. when we update them but some applications like Spotify for instance looses all persistence and settings, login date, ect. when it updates using the scoop installer. How do we handle persistence and is there a method that we could use for apps like Spotify, if it has to do with the portable nature of the bucket then maybe a new manifest for the "non-portable" bucket?

If I knew where to start on the issue I would have already started.

**Comment by rashil2000** [comment, marked as answer]
Spotify loses its data because we run its uninstaller. Maybe we should not run uninstaller during the update process.
/cc @sitiom

**Reply by AdmnJ** [reply]
That makes sense, is there a way to specify a different process for updating then uninstalling and reinstalling in the manifest?

**Reply by rashil2000** [reply]
Inside the manifest we can detect if it's an update command or an uninstall command. Accordingly we can write statements.

**Reply by AdmnJ** [reply]
As far as the spotify package goes I have deduced that removing the AppData\Roaming\Spotify\prefs file is all it takes to wipe an active login.... however that doesn't wipe the settings once you are signed back in so there are more files that should probably be persistent.

**Reply by sitiom** [reply]
Fix on https://github.com/ScoopInstaller/Extras/pull/8535
